tests/query/snql/test_query.py

```diff
@@ -537,12 +611,325 @@
         ),
         id="Basic query with new lines and no ambiguous clause content",
     ),
+    pytest.param(
+        "MATCH (e: events) -[contains]-> (t: Transactions) SELECT 4-5, c",
+        CompositeQuery(
+            from_clause=JoinClause(
+                left_node=IndividualNode(
+                    "e",
+                    QueryEntity(
+                        EntityKey.EVENTS,
+                        get_entity(EntityKey.EVENTS).get_data_model(),
+                        "e",
+                    ),
+                ),
+                right_node=IndividualNode(
+                    "t",
+                    QueryEntity(
+                        EntityKey.TRANSACTIONS,
+                        get_entity(EntityKey.TRANSACTIONS).get_data_model(),
+                        "t",
+                    ),
+                ),
+                keys=[
+                    JoinCondition(
+                        JoinConditionExpression("e", "event_id"),
+                        JoinConditionExpression("t", "event_id"),
+                    )
+                ],
+                join_type=JoinType.INNER,
+                join_modifier=JoinModifier.SEMI,
+            ),
+            selected_columns=[
+                SelectedExpression(
+                    "4-5",
+                    FunctionCall(None, "minus", (Literal(None, 4), Literal(None, 5))),
+                ),
+                SelectedExpression("c", Column("_snuba_c", None, "c")),
```

Comment:
This would be an ambiguous expression (not qualified with the table alias). Are you planning to deal with fully qualifying the columns in a next PR? (please feel free to do it outside of this PR. Just checking in to ensure we are on the same page).

Reply:
Yes I have a note to deal with qualifying columns in a separate PR. Probably an additional post processor to qualify things if possible and fail on ambiguous queries.